pkg/build/controller/strategy/util.go

```diff
@@ -232,9 +232,14 @@ func addSourceEnvVars(source buildapi.BuildSource, output *[]v1.EnvVar) {
 	*output = append(*output, sourceVars...)
 }
 
+// TODO figure out why we provide this information to a pod
```

Comment:
@bparees

Reply:
we compare the origin/master version w/ the version of the binary in the builder image because in an ideal world they should always match and if they don't it's indicative that someone might be doing something weird and that could cause builds to break in weird ways.

it's not the most critical thing in the world so if it's a problem to retain it we can probably get rid of it.

Reply:
> we compare the origin/master version w/ the version of the binary in the builder image because in an ideal world they should always match and if they don't it's indicative that someone might be doing something weird and that could cause builds to break in weird ways.

This is unreliable.  For instance, during upgrades.  Also, as components are split out, the versions will never match again.

In addition, you're using a human readable string as the input, not a machine readable version.

I'd recommend removing it earlier rather than later.

Reply:
actually i'm not even sure that's what we're using this for, i think we get that elsewhere.  this is just information we bake into the images we build as information about what version of the cluster was used to build the image.

Reply:
> actually i'm not even sure that's what we're using this for, i think we get that elsewhere. this is just information we bake into the images we build as information about what version of the cluster was used to build the image.

Removable?  This is going to drift further from reality over time and I'm not sure we'll find it again for a while.

Reply:
yeah i think we can just delete it.

note that there are a few tests that expect to see the ORIGIN_VERSION key so those will need to be cleaned up.